From: MoSunDay <MoSunDay@users.noreply.github.com>
Date: Sat, 17 Oct 2026 12:00:00 +0800
Subject: Disable Periodic Background Sync

Periodic Background Sync lets sites wake the browser in the background on a
schedule defined by the server. Disable it by default as a privacy and CPU
saving measure; the original behaviour can be restored with the
periodic-background-sync flag in chrome://flags.

Need: Add-cromite-flags-support.patch
License: GPL-2.0-or-later - https://spdx.org/licenses/GPL-2.0-or-later.html
---
 .../browser/about_flags_cc/Disable-periodic-background-sync.inc   | 8 ++++++++
 .../blink/common/features_cc/Disable-periodic-background-sync.inc | 1 +
 2 files changed, 9 insertions(+)
 create mode 100644 cromite_flags/chrome/browser/about_flags_cc/Disable-periodic-background-sync.inc
 create mode 100644 cromite_flags/third_party/blink/common/features_cc/Disable-periodic-background-sync.inc

diff --git a/cromite_flags/chrome/browser/about_flags_cc/Disable-periodic-background-sync.inc b/cromite_flags/chrome/browser/about_flags_cc/Disable-periodic-background-sync.inc
new file mode 100644
--- /dev/null
+++ b/cromite_flags/chrome/browser/about_flags_cc/Disable-periodic-background-sync.inc
@@ -0,0 +1,8 @@
+#ifdef FLAG_SECTION
+
+    {"periodic-background-sync", "Disable Periodic Background Sync",
+     "Disable the Periodic Background Sync web feature; sites can no longer wake "
+     "the browser in the background", kOsAll,
+     FEATURE_VALUE_TYPE(blink::features::kPeriodicBackgroundSync)},
+
+#endif
diff --git a/cromite_flags/third_party/blink/common/features_cc/Disable-periodic-background-sync.inc b/cromite_flags/third_party/blink/common/features_cc/Disable-periodic-background-sync.inc
new file mode 100644
--- /dev/null
+++ b/cromite_flags/third_party/blink/common/features_cc/Disable-periodic-background-sync.inc
@@ -0,0 +1 @@
+SET_CROMITE_FEATURE_DISABLED(kPeriodicBackgroundSync);
--
